src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -170,7 +171,7 @@ private static void HandleBinaryOperator(in OperationAnalysisContext context, IN
                     {
                         if (symbolEqualityComparerType != null &&
```

Comment:
Suggestion: collapse the two `if` statements, and move the `symbolEqualityComparerType != null` check before the `TryGetValue` call.

Reply:
I'd rather do the change in a separate PR as it's not really related to this fix.